Add the includes these sources rely on getting transitively through some
Boost.Asio configurations: boost::system::system_error in exception.cpp, and
boost::asio::detail::socket_ops in endpoint.cpp, which Asio only pulls in when
it is not built for separate compilation.

diff --git a/src/detail/exception.cpp b/src/detail/exception.cpp
--- a/src/detail/exception.cpp
+++ b/src/detail/exception.cpp
@@ -9,6 +9,7 @@
 #include <boost/cobalt/error.hpp>
 
 #include <boost/asio/executor.hpp>
+#include <boost/system/system_error.hpp>
 
 namespace boost::cobalt::detail
 {
diff --git a/src/io/endpoint.cpp b/src/io/endpoint.cpp
--- a/src/io/endpoint.cpp
+++ b/src/io/endpoint.cpp
@@ -6,6 +6,7 @@
 //
 
 #include <boost/cobalt/io/endpoint.hpp>
+#include <boost/asio/detail/socket_ops.hpp>
 #include <boost/asio/ip/address.hpp>
 
 namespace boost::cobalt::detail
